cpp/test/sg/rf_depth_test.cu

```diff
@@ -0,0 +1,301 @@
+/*
+ * Copyright (c) 2019-2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <common/cudart_utils.h>
+#include <gtest/gtest.h>
+#include <cuda_utils.cuh>
+#include <cuml/ensemble/randomforest.hpp>
+#include <queue>
+#include <random>
+
+namespace ML {
+
+using namespace MLCommon;
+
+template <typename T>  // template useless for now.
+struct RfInputs {
+  int n_rows;
+  int n_cols;
+  int n_trees;
+  float max_features;
+  float rows_sample;
+  int max_depth;
+  int max_leaves;
+  bool bootstrap;
+  bool bootstrap_features;
+  int n_bins;
+  int split_algo;
+  int min_rows_per_node;
+  float min_impurity_decrease;
+  int n_streams;
+  CRITERION split_criterion;
+};
+
+template <typename T>
+class RfClassifierDepthTest : public ::testing::TestWithParam<int> {
+ protected:
+  void basicTest() {
+    const int max_depth = ::testing::TestWithParam<int>::GetParam();
+    params = RfInputs<T>{5000,
+                         10,
+                         1,
+                         1.0f,
+                         1.0f,
+                         max_depth,
+                         -1,
+                         false,
+                         false,
+                         8,
+                         SPLIT_ALGO::GLOBAL_QUANTILE,
+                         2,
+                         0.0,
+                         2,
+                         CRITERION::ENTROPY};
+
+    DecisionTree::DecisionTreeParams tree_params;
+    set_tree_params(tree_params, params.max_depth, params.max_leaves,
+                    params.max_features, params.n_bins, params.split_algo,
+                    params.min_rows_per_node, params.min_impurity_decrease,
+                    params.bootstrap_features, params.split_criterion, false);
+    RF_params rf_params;
+    set_all_rf_params(rf_params, params.n_trees, params.bootstrap,
+                      params.rows_sample, -1, params.n_streams, tree_params);
+
+    int data_len = params.n_rows * params.n_cols;
+    allocate(data, data_len);
+    allocate(labels, params.n_rows);
+
+    cudaStream_t stream;
+    CUDA_CHECK(cudaStreamCreate(&stream));
+
+    // Populate data (assume Col major)
+    std::mt19937 gen(0);
+    std::vector<T> data_h(data_len);
+    std::normal_distribution<> d{0, 1};
+    for (int col = 0; col < params.n_cols; ++col) {
```

Comment:
In the future, you might find it easier to use the make_blobs primitive in src/datasets